src/prefect/tasks/census/census.py

```diff
@@ -0,0 +1,125 @@
+import time
+
+import re
+import requests
+
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+MIN_WAIT_TIME, DEFAULT_WAIT_TIME = 5, 60
+
+
+class CensusSyncTask(Task):
+    """
+    Task for running Census connector sync jobs.
+
+    This task assumes the user has a Census sync already configured and is attempting to orchestrate the
+    sync using Prefect task to send a post to the API within a prefect flow. Copy and paste from the api
+    trigger section on the configuration page in the `api_trigger` param to set a default sync.
+
+    Args:
+        - api_trigger (str, optional): default sync to trigger, if none is specified in `run`
+        - **kwargs (dict, optional): additional kwargs to pass to the base Task constructor
+    """
+
+    def __init__(self, api_trigger=None, **kwargs):
+        self.api_trigger = api_trigger
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("api_trigger")
+    def run(
+        self, api_trigger: str, poll_status_every_n_seconds: int = DEFAULT_WAIT_TIME
+    ) -> dict:
+        """
+        Task run method for Census syncs.
+
+        An invocation of `run` will attempt to start a sync job for the specified `api_trigger`. `run`
+        will poll Census for the sync status, and will only complete when the sync has completed or
+        when it receives an error status code from the trigger API call.
+
+        Format of api_trigger:
+            - "https://bearer:secret-token:s3cr3t@app.getcensus.com/api/v1/syncs/123/trigger"
+
+        Args:
+            - api_trigger (str): if not specified in run, it will pull from the default for the
+                CensusSyncTask constructor. Keyword argument.
+            - poll_status_every_n_seconds (int, optional): this task polls the Census API for the sync's
+                status. If provided, this value will override the default polling time of
+                60 seconds and it has a minimum wait time of 5 seconds. Keyword argument.
+
+        Returns:
+            - dict: dictionary of statistics returned by Census on the specified sync
+        """
+
+        if not api_trigger:
+            raise ValueError(
+                """Value for parameter `api_trigger` must be provided. See Census sync
+                                configuration page."""
+            )
+
+        pattern = r"https:\/\/bearer:secret-token:(.*)@app.getcensus.com\/api\/v1\/syncs\/(\d*)\/trigger"
```

Comment:
it would be nice if the pattern checking was separated into it's own function

if `api_trigger` is provided at `__init__`, we should check the pattern is correct at that time instead of waiting until run time

Reply:
A miss here is worth a 
```if not confirmed_pattern:
            raise ValueError(
                """Invalid parameter for `api_trigger` please paste directly from the Census
                                sync configuration page. This is CaSe SenSITiVe."""
            )``` 
right?

Reply:
for sure! I'm envisioning something like the following (irrelevant code snippets omitted)

```python
class CensusSyncTask(Task):
    ...
    def __init__(self, api_trigger=None, **kwargs):
        if api_trigger:
            self.confirm_url_pattern(api_trigger)
        self.api_trigger = api_trigger
        super().__init__(**kwargs)

    def run(...):
        ...
        confirmed_pattern = self.get_confirmed_url_pattern(api_trigger)
        secret, sync_id = confirmed_pattern.groups()
        ...

    @staticmethod
    def get_confirmed_url_pattern(api_trigger: str):
        pattern = r"https:\/\/bearer:secret-token:(.*)@app.getcensus.com\/api\/v1\/syncs\/(\d*)\/trigger"
        url_pattern = re.compile(pattern)
        # Making sure it is a valid api trigger.
        confirmed_pattern = url_pattern.match(api_trigger)

        if not confirmed_pattern:
            raise ValueError(
                """Invalid parameter for `api_trigger` please paste directly from the Census
                                sync configuration page. This is CaSe SenSITiVe."""
            )
      return confirmed_pattern

```

Reply:
I didn't see this until I had already pushed the almost identical code (but in the wrong place). Marking this as resolved now.